### ISSUE TITLE
Custom channel names and indexing

### ISSUE THREAD

**Issue by mbauman** [mbauman/Signals.jl#3]
It'd be nice to allow custom channel names for indexing.  Since channels are so often integers, I could see allowing `Union(Symbol, Integer)` for this.  Then a Signal `s` with 3 channels and names `[:snips, 3, 5]` would index like so:

```
s[:snips] -> s.channels[1]
s[3] -> s.channels[2]
s[5] -> s.channels[3]

s[2] -> BoundsError?  Or perhaps there's be a better error type.
```

**Comment by tshort** [comment]
Mixing Ints and Symbols seems confusing to me. It'd be nice to have a way to index to channels by position. I rarely run into channels that are numbered in my area of work. That includes monitoring data from scope-like devices and time series data in R (the zoo package is my favorite there).

Using Ints that way is definitely not in line with the AbstractDataFrame interface.

**Comment by mbauman** [comment]
Yes, I agree.  I don't think I should overload integer indexing anymore.  And for other types, I'm thinking about only supporting intervals as they will generally be Real numbers and shooting for exact equality or doing interpolation under the hood just seems too magical.
